From d12b017cddfeb9fe9920ba054ae3dfcb8e9238b8 Mon Sep 17 00:00:00 2001
From: Florian Weimer <fweimer@redhat.com>
Date: Wed, 4 Mar 2026 18:32:36 +0100
Subject: [PATCH] Linux: Only define OPEN_TREE_* macros in <sys/mount.h> if
 undefined (bug 33921)

There is a conditional inclusion of <linux/mount.h> earlier in the file.
If that defines the macros, do not redefine them.  This addresses build
problems as the token sequence used by the UAPI macro definitions
changes between Linux versions.

Reviewed-by: Adhemerval Zanella  <adhemerval.zanella@linaro.org>
Origin: upstream
Bug: https://sourceware.org/bugzilla/show_bug.cgi?id=33921
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/glibc/+bug/2145679
---
 sysdeps/unix/sysv/linux/sys/mount.h | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

Index: glibc/sysdeps/unix/sysv/linux/sys/mount.h
===================================================================
--- glibc.orig/sysdeps/unix/sysv/linux/sys/mount.h	2026-03-31 15:57:47.729172806 -0400
+++ glibc/sysdeps/unix/sysv/linux/sys/mount.h	2026-03-31 16:00:39.154465840 -0400
@@ -265,8 +265,12 @@
 #define FSOPEN_CLOEXEC          0x00000001
 
 /* open_tree flags.  */
-#define OPEN_TREE_CLONE    1         /* Clone the target tree and attach the clone */
-#define OPEN_TREE_CLOEXEC  O_CLOEXEC /* Close the file on execve() */
+#ifndef OPEN_TREE_CLONE
+# define OPEN_TREE_CLONE    1 /* Clone the target tree and attach the clone */
+#endif
+#ifndef OPEN_TREE_CLOEXEC
+# define OPEN_TREE_CLOEXEC  O_CLOEXEC /* Close the file on execve() */
+#endif
 
 
 __BEGIN_DECLS
